File: Labor6.py
import math

def factorial(num):
    factor = 1
    for k in range (1, num+1):
        factor*=k
        k+=1
    return factor
def cos_x(x, differ):
	i = 2
	k = -1
	previous = 1
	current = k * pow(x, i) / factorial(i)
	cos_x = 1
	while previous-math.fabs(current)>=dif or previous - math.fabs(current)<=0:
		cos_x += current
		previous = math.fabs(current)
		i += 2
		k *= -1
		current = k * pow(x, i)/ factorial(i)
	return cos_x
def answer(m, n, differ):
	y = cos_x(m, differ) + cos_x(n, differ) * cos_x(n, differ) + cos_x((m + n), differ)
	return y
a=float(input("Enter a: "))
b=float(input("Enter b: "))
dif=float(input("Enter accuracy: "))
y=answer(a, b, dif)
print("y=", y)

File: Labor6.cpp
#include <cmath>
#include <iostream>
using namespace std;
double a, b, dif;
long factorial(int);
double cos_x(double);
double answer(double, double);
int main() {
	cout << "Enter a, b and accuracy: ";
	cin >> a >> b>>dif;
	cout << "y=" << answer(a, b); 
	return 0;
}
long factorial(int num) {
	long factor = 1; 
	for (int k = 1; k <= num; k++)
		factor *= k;
	return factor;
}
double cos_x(double x) {
	int i = 2, k = -1;
	double previous = 1;
	double current = k * pow(x, i) / factorial(i);
	double cos_x = 1;
	while (previous-fabs(current)>=dif || previous - fabs(current)<=0) {
		cos_x += current;
		previous = fabs(current);
		i += 2;
		k *= -1;
		current = k * pow(x, i)/ factorial(i);
	}
	return cos_x;
}
double answer(double m, double n) {
	double y = cos_x(m) + cos_x(n) * cos_x(n) + cos_x(m + n);
	return y;
}